src/rgw/rgw_rest_s3.h

```diff
@@ -630,7 +630,7 @@ class RGWHandler_Auth_S3 : public RGWHandler_REST {
     : RGWHandler_REST(),
       auth_registry(auth_registry) {
   }
-  ~RGWHandler_Auth_S3() override = default;
+  virtual ~RGWHandler_Auth_S3() override = default;
```

Comment:
all dtors are trivial. so "virtual" is not strictly necessary.
we can add it to be on the safe side, but then, why ere they removed from src/rgw/rgw_rest_pubsub.h ?

Reply:
removed by accident? AFAICR?   Adding them may look like truth-and-beauty, but at least originally it helped get rid of some undefined refs to the class's vtable.